### PR TITLE
Add TypeScript types to ContentSwitcher, Switch

### LINKED ISSUE #13550: Add TypeScript types to ContentSwitcher, Switch

### PR DESCRIPTION
**Pull request by anton-tsymuk-viacomcbs**
Closes #13550 

Added typescript types to ContentSwitcher, Switch and Layout components.

#### Changelog

**New**

- Added typescript types to ContentSwitcher, Switch and Layout components.

**Changed**

- Updated list of contributors

#### Testing / Reviewing

- Start storybook, check that ContentSwitcher component works as expected.
- Check unit tests

### CODE REVIEW

**Comment by anton-tsymuk-viacomcbs** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/react/src/components/Layout/index.tsx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  /**
+   * Provide a custom class name to be used on the outermost element rendered by
+   * the component
+   */
+  className: PropTypes.string,
+
+  /**
+   * Specify the desired density of components within this layout
+   */
+  density: PropTypes.oneOf(densities),
```
Comment (francinelucca) [inline packages/react/src/components/Layout/index.tsx]:
```suggestion
  density: PropTypes.oneOf<Density>(densities),
```

File: packages/react/src/components/Layout/index.tsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+  /**
+   * Specify the desired size of components within this layout
+   */
+  size: PropTypes.oneOf(sizes),
```
Comment (francinelucca) [inline packages/react/src/components/Layout/index.tsx]:
```suggestion
  size: PropTypes.oneOf<Size>(sizes),
```

File: packages/react/src/components/Layout/index.tsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+} as WeakValidationMap<LayoutProps & React.RefAttributes<React.ReactNode>>;
```
Comment (francinelucca) [inline packages/react/src/components/Layout/index.tsx]: This conversion shouldn't necessary with the suggested changes
```suggestion
}
```

File: packages/react/src/components/Layout/index.tsx [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+import React, {
+  ElementType,
+  HTMLAttributes,
+  ReactNode,
+  WeakValidationMap,
+} from 'react';
+
+import { usePrefix } from '../../internal/usePrefix';
+
+const sizes = ['xs', 'sm', 'md', 'lg', 'xl', '2xl'];
```
Comment (francinelucca) [inline packages/react/src/components/Layout/index.tsx]:
```suggestion
const sizes: Size[]  = ['xs', 'sm', 'md', 'lg', 'xl', '2xl'];
```

File: packages/react/src/components/Layout/index.tsx [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+const densities = ['condensed', 'normal'];
```
Comment (francinelucca) [inline packages/react/src/components/Layout/index.tsx]:
```suggestion
const densities: Density[] = ['condensed', 'normal'];
```

File: packages/react/src/components/Layout/index.tsx [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+  className?: string;
+
+  /**
+   * Specify the desired layout density constraints of this element's children
+   */
+  density?: {
+    min?: Density | null;
+    default?: Density | null;
+    max?: Density | null;
+  };
```
Comment (francinelucca) [inline packages/react/src/components/Layout/index.tsx]:
```suggestion
  density?: {
    min?: Density | null;
    default?: Density | null;
    max?: Density | null;
  } | null;
```

File: packages/react/src/components/Layout/index.tsx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+
+  /**
+   * Specify the desired layout size constraints of this element's children
+   */
+  size?: {
+    min?: Size;
+    default?: Size;
+    max?: Size;
+  };
```
Comment (francinelucca) [inline packages/react/src/components/Layout/index.tsx]:
```suggestion
  size?: {
    min?: Size | null;
    default?: Size | null;
    max?: Size | null;
  } | null;
```

File: packages/react/src/components/Layout/index.tsx [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+  className: PropTypes.string,
+
+  /**
+   * Specify the desired layout density constraints of this element's children
+   */
+  density: PropTypes.shape({
+    min: PropTypes.oneOf(['condensed', 'normal']),
+    default: PropTypes.oneOf(['condensed', 'normal']),
+    max: PropTypes.oneOf(['condensed', 'normal']),
+  }),
```
Comment (francinelucca) [inline packages/react/src/components/Layout/index.tsx]:
```suggestion
  density: PropTypes.shape({
    min: PropTypes.oneOf<Density>(['condensed', 'normal']),
    default: PropTypes.oneOf<Density>(['condensed', 'normal']),
    max: PropTypes.oneOf<Density>(['condensed', 'normal']),
  }),
```

File: packages/react/src/components/Layout/index.tsx [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+
+  /**
+   * Specify the desired layout size constraints of this element's children
+   */
+  size: PropTypes.shape({
+    min: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', '2xl']),
+    default: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', '2xl']),
+    max: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', '2xl']),
+  }),
```
Comment (francinelucca) [inline packages/react/src/components/Layout/index.tsx]:
```suggestion
  size: PropTypes.shape({
    min: PropTypes.oneOf<Size>(['xs', 'sm', 'md', 'lg', 'xl', '2xl']),
    default: PropTypes.oneOf<Size>(['xs', 'sm', 'md', 'lg', 'xl', '2xl']),
    max: PropTypes.oneOf<Size>(['xs', 'sm', 'md', 'lg', 'xl', '2xl']),
  }),
```

File: packages/react/src/components/Layout/index.tsx [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+} as WeakValidationMap<
+  LayoutConstraintProps & React.RefAttributes<React.ReactNode>
+>;
```
Comment (francinelucca) [inline packages/react/src/components/Layout/index.tsx]: This conversion shouldn't be necessary with the suggested changes
```suggestion
}
```

**Review by francinelucca — CHANGES_REQUESTED** [review]
This is looking great @anton-tsymuk-viacomcbs , thanks for getting this PR up!
Just have some cleanup suggestions, let me know your thoughts

**Comment by anton-tsymuk-viacomcbs** [comment]
> This is looking great @anton-tsymuk-viacomcbs , thanks for getting this PR up! Just have some cleanup suggestions, let me know your thoughts

i applied suggested changes. please take a look

**Review by francinelucca — APPROVED** [review]
👏🏽

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**